lib/ansible/modules/windows/win_domain_membership.ps1

```diff
@@ -147,14 +148,14 @@ Function Set-Workgroup {
 
     Write-DebugLog ("Calling JoinDomainOrWorkgroup with workgroup {0}" -f $workgroup_name)
     try {
-        $swg_result = (Get-WmiObject -ClassName Win32_ComputerSystem).JoinDomainOrWorkgroup($workgroup_name)
+        $swg_result = Get-CimInstance Win32_ComputerSystem | Invoke-CimMethod -MethodName JoinDomainOrWorkgroup -Arguments @{Name="$workgroup_name"}
```

Comment:
Not sure if this should be part of this PR. May need to be discussed separately.

Reply:
This was one of the problems with pslint and it is the same implementation just with CIM without any functional change

Reply:
As I said, I am not sure. Someone else has to verify this.